server/src/routines/virtual/new-boss-ai.ts

```diff
@@ -987,7 +987,7 @@ function getNextTurnAction(
 		return [
 			{
 				type: selectedAttackType,
-			},
+			} as AttackActionData,
```

Comment:
```suggestion
			} satisfies AttackActionData,
```
`as` overrides the type regardless of value, `satisfies` asserts the value does match the provided type